MANIFEST.in

```diff
@@ -0,0 +1,18 @@
+recursive-exclude .github/
+recursive-exclude docs/
+exclude .readthedocs.yaml
+exclude .gitignore
+
+recursive-include test/
```

Comment:
The [recommended approach](https://setuptools.pypa.io/en/latest/userguide/miscellaneous.html) - to include things broadly and then exclude them. The excludes remove from the collected list so need to be after the includes (I had this wrong originally).

This is mostly a leftover from the earlier version of the PR where I was going to use setuptools-scm for the version. I'm not sure we want to bother with this now. The difference between the default of not having a MANIFEST.in and having the current one is just this:

```
Only in implicit/ops-2.10.0.dev0: docs
Only in implicit/ops-2.10.0.dev0: .github
Only in explicit/ops-2.10.0.dev0/test/charms/test_main/lib: ops
```

The last one is interesting - it's not a file, it's a link. The default behaviour is that it's just skipped, but if it's included then it actually follows the link and there's a second copy of `ops` in that location in the tar.gz. Neither of these seem correct, but the latter seems a bit better in that I assume everything would still work for the test in that case, whereas it would not with it missing.

I feel that if you get the sdist then you're not looking for the `.github` folder, or the content of the `docs` folder. But it's also not much of an issue to have them there and the release bundle that Github makes presumably includes them.

I lend towards keeping the MANIFEST.in as it is now, but am open to arguments that we just get rid of it.